test/unit/org/apache/cassandra/config/DataRateSpecTest.java

```diff
@@ -79,38 +99,94 @@ public void testFromSymbol()
+                                                                                                        .hasMessageContaining("Invalid data rate: 9223372036854775802KiB/s. " +
+                                                                                                 "It shouldn't be more than 9223372036854775806 in bytes_per_second");
+    }
+
+    @Test
+    public void testValidUnits()
+    {
+        // we need toString as internally it is double and they are not 0.0 equal but for the end user the double numbers don't exist
+        assertEquals(new DataRateSpec.IntMebibytesPerSecondBound("24MiB/s").toString(), DataRateSpec.IntMebibytesPerSecondBound.megabitsPerSecondInMebibytesPerSecond(200).toString());
     }
 
     @Test
     public void testEquals()
     {
-        assertEquals(new DataRateSpec("10B/s"), new DataRateSpec("10B/s"));
-        assertEquals(new DataRateSpec("10KiB/s"), new DataRateSpec("10240B/s"));
-        assertEquals(new DataRateSpec("10240B/s"), new DataRateSpec("10KiB/s"));
-        assertEquals(DataRateSpec.inMebibytesPerSecond(Long.MAX_VALUE), DataRateSpec.inMebibytesPerSecond(Long.MAX_VALUE));
-        assertNotEquals(DataRateSpec.inMebibytesPerSecond(Long.MAX_VALUE), DataRateSpec.inBytesPerSecond(Long.MAX_VALUE));
-        assertNotEquals(new DataRateSpec("0KiB/s"), new DataRateSpec("10MiB/s"));
+        assertEquals(new DataRateSpec.LongBytesPerSecondBound("10B/s"), new DataRateSpec.LongBytesPerSecondBound("10B/s"));
+        assertEquals(new DataRateSpec.LongBytesPerSecondBound("10KiB/s"), new DataRateSpec.LongBytesPerSecondBound("10240B/s"));
+        assertEquals(new DataRateSpec.LongBytesPerSecondBound("10240B/s"), new DataRateSpec.LongBytesPerSecondBound("10KiB/s"));
+        assertNotEquals(new DataRateSpec.LongBytesPerSecondBound("0KiB/s"), new DataRateSpec.LongBytesPerSecondBound("10MiB/s"));
+
+        assertEquals(new DataRateSpec.IntMebibytesPerSecondBound("10B/s"), new DataRateSpec.IntMebibytesPerSecondBound("10B/s"));
+        assertEquals(new DataRateSpec.IntMebibytesPerSecondBound("10KiB/s"), new DataRateSpec.IntMebibytesPerSecondBound("10240B/s"));
+        assertEquals(new DataRateSpec.IntMebibytesPerSecondBound("10240B/s"), new DataRateSpec.IntMebibytesPerSecondBound("10KiB/s"));
+        assertNotEquals(new DataRateSpec.IntMebibytesPerSecondBound("0KiB/s"), new DataRateSpec.IntMebibytesPerSecondBound("10MiB/s"));
+
+        assertEquals(new DataRateSpec.IntMebibytesPerSecondBound("10B/s"), new DataRateSpec.LongBytesPerSecondBound("10B/s"));
+        assertEquals(new DataRateSpec.IntMebibytesPerSecondBound("10KiB/s"), new DataRateSpec.LongBytesPerSecondBound("10240B/s"));
+        assertEquals(new DataRateSpec.IntMebibytesPerSecondBound("10240B/s"), new DataRateSpec.LongBytesPerSecondBound("10KiB/s"));
+        assertNotEquals(new DataRateSpec.IntMebibytesPerSecondBound("0KiB/s"), new DataRateSpec.LongBytesPerSecondBound("10MiB/s"));
```

Comment:
We can mark the method with `@SuppressWarnings("AssertEqualsBetweenInconvertibleTypes")` to get rid of the warning.